docs/next.config.js

```diff
@@ -21,7 +21,7 @@ module.exports = withDocsInfra({
     FEEDBACK_URL: process.env.FEEDBACK_URL,
     CONTEXT: process.env.CONTEXT,
     // #default-branch-switch
-    SOURCE_CODE_ROOT_URL: 'https://github.com/mui/mui-x/blob/master',
+    DEFAULT_BRANCH: 'master',
```

Comment:
You probably need to use the same name as in the `monorepo` to avoid getting undefined in the generated `Edit this page` link. 🤔 
```suggestion
    SOURCE_GITHUB_BRANCH: 'master',
```

Reply:
Thanks for the catch up, I did not noticed the file diff I was looking to was outdated 

![image](https://github.com/mui/mui-x/assets/45398769/a1188251-a865-4486-acf5-bd21f0ecaf93)